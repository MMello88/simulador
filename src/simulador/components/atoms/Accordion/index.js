/**
 import Accordion from "./Accordion"
 export default Accordion

 ou a expressão simplificada abaixo
 */
export { default } from "./Accordion";
export { default as AccordionGroup } from "./AccordionGroup";


/**
 * ao montar esta condição do index e seu servidor estiver online e
 * recerber este update, vc precisa restartar o servico para ele enterder 
 * esta mudança de comportamento para ler o index.js
 */
