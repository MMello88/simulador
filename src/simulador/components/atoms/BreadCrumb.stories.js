import React from "react";

import BreadCrumb from "simulador/components/atoms/BreadCrumb";

export default {
  title: "Components/Atoms/BreadCrumb",
  component: BreadCrumb,
}

const items = [
  {label: "Inicio", link: "/"},
  {label: "Sobre", link: ""}
]

export const usege = () => (
  <BreadCrumb items={items} />
);