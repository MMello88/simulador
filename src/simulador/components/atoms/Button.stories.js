import React from 'react';
import {actions} from "@storybook/addon-actions";
import Button from 'simulador/components/atoms/Button';
import styled from 'styled-components'


export default {
  title: "Components/Atoms/Button",
  component: Button
}
const Toolbar = styled.div`
  & > * {
    margin-right: 8px;
  }
`;

const events = actions({onClick: "cliked"});

export const usage = () => (
  <div>
    <p>Enabled:</p>
    <Toolbar>
      <Button {...events}>Default</Button>
      <Button {...events} color="primary" >Primary</Button>
      <Button {...events} color="danger" >Danger</Button>
    </Toolbar>
    <p>Disabled:</p>
    <Toolbar>
      <Button disabled {...events}>Default</Button>
      <Button disabled {...events} color="primary" >Primary</Button>
      <Button disabled {...events} color="danger" >Danger</Button>
    </Toolbar>
  </div>
  
);

export const outlined = () => (
  <div>
    <p>Enabled:</p>
    <Toolbar>
      <Button {...events} variant="outlined" >Default</Button>
      <Button {...events} variant="outlined" color="primary" >Primary</Button>
      <Button {...events} variant="outlined" color="danger" >Danger</Button>
    </Toolbar>
    <p>Disabled:</p>
    <Toolbar>
      <Button disabled {...events} variant="outlined" >Default</Button>
      <Button disabled {...events} variant="outlined" color="primary" >Primary</Button>
      <Button disabled {...events} variant="outlined" color="danger" >Danger</Button>
    </Toolbar>
  </div>
  
);

export const link = () => (
  <div>
    <p>Enabled:</p>
    <Toolbar>
      <Button {...events} variant="link" >Default</Button>
      <Button {...events} variant="link" color="primary" >Primary</Button>
      <Button {...events} variant="link" color="danger" >Danger</Button>
    </Toolbar>
    <p>Disabled:</p>
    <Toolbar>
      <Button disabled {...events} variant="link" >Default</Button>
      <Button disabled {...events} variant="link" color="primary" >Primary</Button>
      <Button disabled {...events} variant="link" color="danger" >Danger</Button>
    </Toolbar>
  </div>
  
);
