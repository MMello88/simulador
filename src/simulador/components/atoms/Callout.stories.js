import React from "react";

import Callout, {CalloutBody, CalloutActions, CalloutMedia} from "simulador/components/atoms/Callout";
import Button from "simulador/components/atoms/Button";
import SpeedImage from "draws/Speed"

export default {
  title: "Components/Atoms/Callout",
  component: Callout,
}

export const usege = () => (
  <Callout>
    <CalloutBody>
      <h6>Ex commodo nulla dolor sit sunt veniam dolor mollit.</h6>
      <p>Cillum incididunt labore elit ut labore reprehenderit fugiat duis aute minim proident.</p>

      <CalloutActions>
        <Button color="primary" >Matricular</Button>
      </CalloutActions>
      
    </CalloutBody>
    
    <CalloutMedia>
      <SpeedImage></SpeedImage>
    </CalloutMedia>


    
  </Callout>
);