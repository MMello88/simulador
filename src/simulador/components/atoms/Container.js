import styled from 'styled-components';

import { breakAt, BreakpointSize } from 'simulador/styles/Breakpoints';

const Container = styled.div`
  width: 100%;
  padding: 0 8px;

  ${breakAt(BreakpointSize.sm)}{
    padding: 0 16px;
  }

  ${breakAt(BreakpointSize.lg)}{
    padding: 0;
    max-width: 1140px;
    margin: 0 auto;
  }
`;

export default Container;
