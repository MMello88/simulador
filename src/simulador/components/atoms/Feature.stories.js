import React from "react";
import { FaCar, FaKey, FaMapMarkedAlt, FaAccessibleIcon } from "react-icons/fa";

import Feature from "simulador/components/atoms/Feature";
import Grid from "simulador/components/atoms/Grid";

export default {
  title: "Components/Atoms/Feature ", 
  component: Feature,
}

export const usege = () => (
  <Feature title="title test" icon={<FaCar />}>
    <p>Ex elit officia laboris nulla quis consectetur commodo.</p>
  </Feature>
);

export const withGrid = () => (
  <Grid sm={2} md={3} lg={4} xl={5}>
    <Feature title="Maior e Melhor frota" icon={<FaCar />}>
      <p>Ex elit officia laboris nulla quis consectetur commodo.</p>
    </Feature>
    <Feature title="Habilitação mais rapida" icon={<FaKey />}>
      <p>Ex elit officia laboris nulla quis consectetur commodo.</p>
    </Feature>
    <Feature title="Excelente localização" icon={<FaMapMarkedAlt />}>
      <p>Ex elit officia laboris nulla quis consectetur commodo.</p>
    </Feature>
    <Feature title="Acessibilidade" icon={<FaAccessibleIcon />}>
      <p>Ex elit officia laboris nulla quis consectetur commodo.</p>
    </Feature>
  </Grid>
);
