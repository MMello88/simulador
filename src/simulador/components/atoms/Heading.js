import styled from 'styled-components';

import { BreakpointSize, breakAt } from 'simulador/styles/Breakpoints'

export const Heading = styled.div`
  h1, h2, h3, h4, h5, h6 {
    line-height: 1.3;
    font-weight: 300;

    position: relative;
    margin-bottom: 25px;
    padding-bottom: 8px;
    border-bottom: 1px solid rgba(255, 255, 255, 0.2);
    
    &::after{
      content: "";
      position: absolute;
      left: 0;
      bottom: -3px;
      background: ${(props) => props.theme.colors.primary.main};
      height: 5px;
      width: 70px;
    }

    strong {
      color:  ${(props) => props.theme.colors.primary.main};
    }
  }
  
  h1 {
    padding-bottom: 25px;
  }

  h1 {
    font-size: 2.5rem;

    ${breakAt(BreakpointSize.lg)} {
      font-size: 3.75rem;
    }
  }

  h2 {
    font-size: 2rem;
    ${breakAt(BreakpointSize.lg)} {
      font-size: 3.125rem;
    }
  }

  h3 {
    font-size: 1.9rem;
    ${breakAt(BreakpointSize.lg)} {
      font-size: 2.5rem;
    }
  }

  h4 {
    font-size: 1.3rem;
    font-weight: 600;
    ${breakAt(BreakpointSize.lg)} {
      font-size: 2.125rem;
    }
  }
  
  h5 {
    font-size: 1.2rem;
    font-weight: 300;
    ${breakAt(BreakpointSize.lg)} {
      font-size: 1.5rem;
    }
  }  

  h6 {
    font-size: 1.1rem;
    font-weight: 600;
    ${breakAt(BreakpointSize.lg)} {
      font-size: 1.25rem;
    }
  }
`;
