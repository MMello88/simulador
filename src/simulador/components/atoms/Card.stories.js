import React from "react";

import Card, {CardBody, CardMedia} from "simulador/components/atoms/Card";
import Section from "simulador/components/molecules/Section";
import { Heading } from "simulador/components/atoms/Heading";
import Button from "simulador/components/atoms/Button";

import car1 from "simulador/assets/car1.jpg";
/*import car2 from "simulador/assets/car2.jpg";
import car3 from "simulador/assets/car3.jpg";*/

export default {
  title: "Components/Atoms/Card",
  component: Card,
  subcomponents: {CardBody, CardMedia}
}

export const usege = () => (
  <Section inverse>
    <Card>
      <CardBody>
        <Heading>
          <h6>Título</h6>
        </Heading>
        <p>Mollit magna sunt aute excepteur voluptate cupidatat dolor.</p>
        <div>
          <Button color="primary" variant="link">Saiba mais</Button>
        </div>
      </CardBody>
    </Card>
  </Section>
);


export const withMedia = () => (
  <Section inverse>
    <Card>
      <CardMedia image={car1}></CardMedia>
      <CardBody>
        <Heading>
          <h6>Título</h6>
        </Heading>
        <p>Mollit magna sunt aute excepteur voluptate cupidatat dolor.</p>
        <div>
          <Button color="primary" variant="link">Saiba mais</Button>
        </div>
      </CardBody>
    </Card>
  </Section>
);
