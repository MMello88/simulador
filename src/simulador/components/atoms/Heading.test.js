import React from 'react';
import { render } from 'test-utils';

import { Heading } from 'simulador/components/atoms/Heading';

test('match snapshot', () => {
  const {asFragment}  = render(<Heading>Title</Heading>);

  expect(asFragment()).toMatchSnapshot();
});
