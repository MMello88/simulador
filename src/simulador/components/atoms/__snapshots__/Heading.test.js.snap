// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<DocumentFragment>
  .c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  line-height: 1.3;
  font-weight: 300;
  position: relative;
  margin-bottom: 25px;
  padding-bottom: 8px;
  border-bottom: 1px solid rgba(255,255,255,0.2);
}

.c0 h1::after,
.c0 h2::after,
.c0 h3::after,
.c0 h4::after,
.c0 h5::after,
.c0 h6::after {
  content: "";
  position: absolute;
  left: 0;
  bottom: -3px;
  background: #ffc107;
  height: 5px;
  width: 70px;
}

.c0 h1 strong,
.c0 h2 strong,
.c0 h3 strong,
.c0 h4 strong,
.c0 h5 strong,
.c0 h6 strong {
  color: #ffc107;
}

.c0 h1 {
  padding-bottom: 25px;
}

.c0 h1 {
  font-size: 2.5rem;
}

.c0 h2 {
  font-size: 2rem;
}

.c0 h3 {
  font-size: 1.9rem;
}

.c0 h4 {
  font-size: 1.3rem;
  font-weight: 600;
}

.c0 h5 {
  font-size: 1.2rem;
  font-weight: 300;
}

.c0 h6 {
  font-size: 1.1rem;
  font-weight: 600;
}

@media (min-width:992px) {
  .c0 h1 {
    font-size: 3.75rem;
  }
}

@media (min-width:992px) {
  .c0 h2 {
    font-size: 3.125rem;
  }
}

@media (min-width:992px) {
  .c0 h3 {
    font-size: 2.5rem;
  }
}

@media (min-width:992px) {
  .c0 h4 {
    font-size: 2.125rem;
  }
}

@media (min-width:992px) {
  .c0 h5 {
    font-size: 1.5rem;
  }
}

@media (min-width:992px) {
  .c0 h6 {
    font-size: 1.25rem;
  }
}

<div
    class="c0"
  >
    Title
  </div>
</DocumentFragment>
`;
